snuba/web/rpc/v1/trace_item_attribute_values.py

```diff
@@ -0,0 +1,126 @@
+    )
+
+    truncate_request_meta_to_day(request.meta)
+
+    res = Query(
+        from_clause=entity,
+        selected_columns=[
+            SelectedExpression(
+                name="attr_value",
+                expression=f.distinct(column("attr_value", alias="attr_value")),
+            ),
+        ],
+        condition=base_conditions_and(
+            request.meta,
+            f.equals(column("attr_key"), literal(request.key.name)),
+            # multiSearchAny has special treatment with ngram bloom filters
+            # https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree#functions-support
+            f.multiSearchAny(
+                column("attr_value"),
+                literals_array(None, [literal(request.value_substring_match)]),
+            ),
+        )
+        if request.value_substring_match is not None
+        else base_conditions_and(
+            request.meta,
+            f.equals(column("attr_key"), literal(request.key.name)),
+        ),
+        order_by=[
+            OrderBy(
+                direction=OrderByDirection.ASC, expression=column("organization_id")
+            ),
+            OrderBy(direction=OrderByDirection.ASC, expression=column("attr_key")),
+            OrderBy(direction=OrderByDirection.ASC, expression=column("attr_value")),
+        ],
+        limit=request.limit,
+    )
+    treeify_or_and_conditions(res)
+    return res
+
+
+def _build_snuba_request(
+    request: TraceItemAttributeValuesRequest,
+) -> SnubaRequest:
+    return SnubaRequest(
+        id=uuid.uuid4(),
+        original_body=MessageToDict(request),
+        query=_build_query(request),
+        query_settings=HTTPQuerySettings(),
+        attribution_info=AttributionInfo(
+            referrer=request.meta.referrer,
+            team="eap",
+            feature="eap",
+            tenant_ids={
+                "organization_id": request.meta.organization_id,
+                "referrer": request.meta.referrer,
+            },
+            app_id=AppID("eap"),
+            parent_api="trace_item_values",
+        ),
+    )
+
+
+class AttributeValuesRequest(
+    RPCEndpoint[TraceItemAttributeValuesRequest, TraceItemAttributeValuesResponse]
+):
+    @classmethod
+    def version(cls) -> str:
+        return "v1"
+
+    @classmethod
+    def request_class(cls) -> Type[TraceItemAttributeValuesRequest]:
+        return TraceItemAttributeValuesRequest
+
+    def _execute(
+        self, in_msg: TraceItemAttributeValuesRequest
+    ) -> TraceItemAttributeValuesResponse:
+        snuba_request = _build_snuba_request(in_msg)
+        res = run_query(
+            dataset=PluggableDataset(name="eap", all_entities=[]),
+            request=snuba_request,
+            timer=self._timer,
+        )
+        return TraceItemAttributeValuesResponse(
+            values=[r["attr_value"] for r in res.result.get("data", [])],
+            page_token=PageToken(offset=in_msg.page_token.offset),
```

Comment:
This is wrong.

Reply:
Hmm I'm having trouble understanding this part. I thought right now we're okay with using offset as a basic functionality

Reply:
Hmm I'm not sure if I understand this part. I thought we're okay with using offset as a basic functionality for now.

Reply:
Yes and what are you passing into the offset

Reply:
and did you test the pagetoken functinality?